library/ssl_tls13_generic.c

```diff
@@ -4643,12 +4643,14 @@ int mbedtls_ssl_parse_new_session_ticket( mbedtls_ssl_context *ssl )
     msg = ssl->in_msg + mbedtls_ssl_hs_hdr_len( ssl );
     msg_len = ( ssl->in_msg[2] << 8 ) | ssl->in_msg[3];
 
-    if( msg_len+ mbedtls_ssl_hs_hdr_len( ssl ) != ssl->in_msglen )
+    if( ssl->in_hslen == 0 || ssl->in_hslen > ssl->in_msglen )
```

Comment:
Why are we re-calculating `msg_len` in L.4644? Isn't the length of the HS message stores in `in_hslen` already? Likewise, since we don't support HS fragmentation for now, the messaging layer will have checked that `ssl->in_hslen` isn't greater than `ssl->in_msglen`.